network.go

```diff
@@ -931,7 +931,7 @@ func (n *network) ipamAllocateVersion(ipVer int, ipam ipamapi.Ipam) error {
 	}
 
 	if len(*cfgList) == 0 {
-		if ipVer == 6 {
+		if ipVer == 6 && !n.enableIPv6 {
```

Comment:
I think this check should instead take into account whether the ipam driver supports auto-allocation of ipv6 pools.
Now that `ipamapi.Capability` is in master, I'd suggest to add a `SupportsAutoIPv6` bool to the ipam driver capability.

In fact the reason why this function was bailing out in case of empty `ipamV6Config` was just because the in-built ipam driver does not have the capability of self assigning an ipv6 pool from some predefined list.